src/components/Tree/Tree.tsx

```diff
@@ -129,16 +139,13 @@ const reducer = produce((draft: TreeState, action: TreeStateAction) => {
                     return;
                 }
 
+                const offset = draft.nodes.findLastIndex((node) => node.props.parentId === id) - sliceIndex;
```

Comment:
Note that when the item is deleted `node.props.parentId` still contains the old data, so I think this won't exclude it from the list.
That's why I added the filter part:
```
((node.props.parentId === id && newNodeChildrenIds.includes(node.props.id)) || node.props.parentId !== id)
```

Reply:
nvm... I see what you did... it might work 😄